babel/util.py

```diff
@@ -205,6 +206,24 @@ class TextWrapper(textwrap.TextWrapper):
         r'(?<=[\w\!\"\'\&\.\,\?])-{2,}(?=\w))',   # em-dash
     )
 
+    # e.g. '\u2068foo bar.py\u2069:42'
+    _enclosed_filename_re = re.compile(r'(\u2068[^\u2068]+?\u2069(?::-?\d+)?)')
+
+    def _split(self, text):
+        """Splits the text into indivisible chunks while ensuring that file names
+        containing spaces are not broken up.
+        """
+        enclosed_filename_start = '\u2068'
+        if enclosed_filename_start not in text:
+            # There are no file names which contain spaces, fallback to the default implementation
+            return super()._split(text)
+
+        super_ = super() #  Python 3.9 fix, super() does not work in list comprehension
+        chunks = re.split(self._enclosed_filename_re, text)
+        chunks = [[c] if c.startswith(enclosed_filename_start) else super_._split(c) for c in chunks]
+        chunks = [c for c in chain.from_iterable(chunks) if c]
+        return chunks
+
 
 def wraptext(text: str, width: int = 70, initial_indent: str = '', subsequent_indent: str = '') -> list[str]:
```

Comment:
The documentation of `wraptext()` should be updated to reflect the new behaviour (it says "This version does not wrap lines on hyphens in words.")

Reply:
Also, a little perf thing – maybe for this PR, maybe for a subsequent one: `wraptext()` is called repeatedly in `pofile.py` – it would be more efficient to construct two different `TextWrapper`s in advance and call `wrap()` on them.

I think `babel.util.wraptext()` should probably be deprecated then.

I can whip up a PR to do that...

Reply:
> The documentation of wraptext() should be updated to reflect the new behaviour (it says "This version does not wrap lines on hyphens in words.")

This version doesn't wrap lines on hyphens either, unless I misunderstood your comment? Did you mean that we should add a comment about filenames containing spaces?

Reply:
Yeah, I meant adding a comment about the new filenames-with-spaces behavior.

Reply:
Comment added!